src/snowflake/snowpark/_internal/xml_reader.py

```diff
@@ -52,6 +52,45 @@ def get_file_size(filename: str) -> Optional[int]:
         return file_obj.tell()
 
 
+def tag_is_self_closing(
+    file_obj: Union[BinaryIO, SnowflakeFile],
```

Comment:
If I understand correctly, the file pointer may not be at the beginning when this function is called, right? But this function has a side-effect of incrementing the pointer. Would it be better if we go back to the older approach with has start file position as input? Seems like that approach will be easier to have a no side-effect implementation

Reply:
Actually we expect to increment the pointer here, similar to `find_next_closing_tag_pos`. Because later we will start here to find the next closing tag:
```
  try:
                is_self_close, tag_end = tag_is_self_closing(f)
            except EOFError:
                # malformed XML record
                break

            if is_self_close:
                record_end = tag_end
            else:
                f.seek(tag_end)
                try:
                    record_end = find_next_closing_tag_pos(f, closing_tag, chunk_size)
                except EOFError:
                    # incomplete XML record
                    break
```

Reply:
I can add a comment for this function that will increment the pointer.

Reply:
done